File: src/main/java/org/example/attendTrack/site/Site.java
package org.example.attendTrack.site;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.UUID;

@Entity
@Table(name = "sites")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Site {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    private String address;

    @Column(nullable = false)
    private double lat;

    @Column(nullable = false)
    private double lng;

    @Column(nullable = false)
    @Builder.Default
    private int radiusMeters = 200;

    @Column(name = "work_start_time")
    private LocalTime workStartTime;

    @Column(name = "work_end_time")
    private LocalTime workEndTime;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    public void update(String name, String address, double lat, double lng,
                       int radiusMeters, LocalTime workStartTime, LocalTime workEndTime) {
        this.name = name;
        this.address = address;
        this.lat = lat;
        this.lng = lng;
        this.radiusMeters = radiusMeters;
        this.workStartTime = workStartTime;
        this.workEndTime = workEndTime;
    }

    public void deactivate() {
        this.active = false;
    }
}

File: frontend/types/site.ts
import type { UserResponse } from './user'

export interface CheckpointResponse {
  id: string
  name: string | null
  lat: number
  lng: number
  radiusMeters: number
}

export interface CheckpointRequest {
  id?: string | null
  name?: string | null
  lat: number
  lng: number
  radiusMeters: number
}

export interface SiteResponse {
  id: string
  name: string
  address: string | null
  lat: number
  lng: number
  radiusMeters: number
  workStartTime: string | null  // "HH:mm:ss"
  workEndTime: string | null
  active: boolean
  createdAt: string
  managers: UserResponse[]
  workers: UserResponse[]
  checkpoints: CheckpointResponse[]
}

export interface SiteRequest {
  name: string
  address?: string
  lat: number
  lng: number
  radiusMeters?: number
  workStartTime?: string | null  // "HH:mm:ss"
  workEndTime?: string | null
  checkpoints?: CheckpointRequest[]
  companyId?: string  // required on create
}
